### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName false-positive for non-static nested classes

### ISSUE THREAD

**Issue by krichter722** [pmd/pmd#2029]
**Affects PMD Version:** 6.18.0

**Rule:** [UnnecessaryFullyQualifiedName](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname)

**Description:**
`UnnecessaryFullyQualifiedName` shouldn't apply on right-hand side of instanceof

**Code Sample demonstrating the issue:**
Given
```java
package de.richtercloud.pmd.importt.qualifier.clazz;

public class SomeOtherClass {

    public static void someMethod() {
        System.out.println("someMethod");
    }

    protected class SomeInnerClass {

        public void alsoDoSomething() {
            System.out.println("alsoDoSomething");
        }
    }
}
```
and
```java
package de.richtercloud.pmd.importt.qualifier.clazz;

import static de.richtercloud.pmd.importt.qualifier.clazz.SomeOtherClass.*;

public class SomeClass {

    public void theMethod() {
        someMethod();
        Object someObject = new Object();
        if(someObject instanceof SomeOtherClass.SomeInnerClass) {
            System.out.println("");
        }
    }
}
```
PMD fails due to
```
[INFO] PMD Failure: de.richtercloud.pmd.importt.qualifier.clazz.SomeClass:10 Rule:UnnecessaryFullyQualifiedName Priority:4 Unnecessary use of fully qualified name 'SomeOtherClass.SomeInnerClass' due to existing static import 'de.richtercloud.pmd.importt.qualifier.clazz.SomeOtherClass.*'.
```
This failure is incorrect because omitting `SomeOtherClass.` from `instanceof SomeOtherClass.SomeInnerClass` causes the code to become uncompilable because the inner class name can't be resolved.

An SSCCE is provided at https://gitlab.com/krichter/pmd-import-qualifier-class.

**Running PMD through:** *Maven*

**Comment by adangel** [comment]
Thanks for the report and the test case!

> causes the code to become uncompilable because the inner class name can't be resolved.

That's because the static import only imports static, and the inner class is not static.

The rule doesn't seem to distinguish between static/non-static members here, which causes this false positive.

**Comment by adangel** [comment]
> UnnecessaryFullyQualifiedName also incorrectly applies to methods whose name is covered by a static import, but whose name is used in the class where they're used as well. This leads to a failure to resolve the name. I'll raise a separate issue after this one is fix since you might fix it accidentally as well.

Would you mind creating an issue for this now? Sample code would be helpful.

**Comment by krichter722** [comment]
> Would you mind creating an issue for this now? Sample code would be helpful.

I cannot reproduce this because the situation I'm describing doesn't compile. I should have done it when I created this issue. I'll create a separate issue as soon as I can reproduce it.
